extensions/chghost-3.2.md

```diff
@@ -5,59 +5,87 @@ copyrights:
   -
     name: "Christine Dodrill"
     period: "2013"
-    email: "xena@yolo-swag.com"
+    email: "me@christine.website"
+  -
+    name: "Ryan"
+    period: "2016"
+    email: "ryan@hashbang.sh"
 ---
-The chghost client capability allows a server to directly inform clients about a
-host or user change without having to send a fake quit and join. This capability
-MUST be referred to as `chghost` at capability negotiation time.
 
-When enabled, clients will get the CHGHOST message to designate the host of a
-user changing for clients on common channels with them.
+## Introduction
+
+The `chghost` client capability allows servers to directly inform clients about
+clients changing their host or user without having to simulate the client
+reconnecting. This is useful for servers implementing virtual hosts or masks
+and helps reduce clutter on the client's UI.
+
+## The `chghost` capability
+
+When a client changes their user or host, servers MUST send the `CHGHOST`
+message to clients who have enabled the `chghost` capability and either share
+the same channel as the target client or have the client on a `MONITOR` list.
+Servers SHOULD additionally send the `CHGHOST` message to the client whose
+user or host has changed if the client supports the `chghost` capability.
 
-The CHGHOST message is one of the following:
+When the server sends a `CHGHOST` message to a client, it MUST defer doing so
+until both successful `NICK` and `USER` commands have been received by the
+server.  The server MAY choose to defer it until after registration is
```

Comment:
I find this form (when you do X, you must defer doing X) pretty confusing.

Reply:
Yeah I want to rewrite a bunch of this.